File: backend/app/schemas/calendar.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class CalendarConnectResponse(BaseModel):
    auth_url: str


class CalendarCallbackInput(BaseModel):
    code: str
    state: str


class CalendarConnectionRead(BaseModel):
    id: int
    provider: str
    account_email: str | None = None
    status: str
    last_synced_at: datetime | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CalendarStatusResponse(BaseModel):
    connected: bool
    connection: CalendarConnectionRead | None = None


class CalendarEventRead(BaseModel):
    id: int
    calendar_connection_id: int
    workspace_id: int
    user_id: int
    external_event_id: str
    title: str
    description: str | None = None
    start_time: datetime
    end_time: datetime
    timezone: str | None = "UTC"
    organizer_email: str | None = None
    attendees_json: str | None = None
    meeting_url: str | None = None
    status: str
    meeting_id: int | None = None
    synced_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class UpcomingEventsResponse(BaseModel):
    items: list[CalendarEventRead]
    total: int
    page: int
    size: int


class CalendarSyncResponse(BaseModel):
    message: str
    synced_events_count: int
    created_meetings_count: int

File: frontend/src/types/calendar.ts
export interface CalendarConnectionRead {
  id: number;
  provider: string;
  account_email?: string | null;
  status: string;
  last_synced_at?: string | null;
  created_at: string;
}

export interface CalendarStatusResponse {
  connected: boolean;
  connection?: CalendarConnectionRead | null;
}

export interface CalendarEventRead {
  id: number;
  calendar_connection_id: number;
  workspace_id: number;
  user_id: number;
  external_event_id: string;
  title: string;
  description?: string | null;
  start_time: string;
  end_time: string;
  timezone?: string | null;
  organizer_email?: string | null;
  attendees_json?: string | null;
  meeting_url?: string | null;
  status: string;
  meeting_id?: number | null;
  synced_at?: string | null;
}

export interface UpcomingEventsResponse {
  items: CalendarEventRead[];
  total: number;
  page: number;
  size: number;
}

export interface CalendarSyncResponse {
  message: string;
  synced_events_count: number;
  created_meetings_count: number;
}
